browser/admin/src/integrator/AdminIntegratorSettings.ts

```diff
@@ -417,6 +473,535 @@ class SettingIframe {
+		const labelText = _(this.settingLabels[key] || key);
+
+		return this.createCheckbox(
+			uniqueId,
+			value,
+			labelText,
+			(inputCheckbox, checkboxWrapper, materialIconContainer) => {
+				const currentChecked = !inputCheckbox.checked;
+				inputCheckbox.checked = currentChecked;
+				checkboxWrapper.classList.toggle(
+					'checkbox-radio-switch--checked',
+					!currentChecked,
+				);
+				materialIconContainer.innerHTML = currentChecked
+					? this.SVG_ICONS.checkboxMarked
+					: this.SVG_ICONS.checkboxBlankOutline;
+				data[key] = currentChecked;
+			},
+		);
+	}
+
+	private collectBrowserSettingsFromUI(
+		browserSettingSection: HTMLElement,
+	): void {
+		const inputs = browserSettingSection.querySelectorAll<HTMLInputElement>(
+			'input.checkbox-radio-switch-input',
+		);
+
+		inputs.forEach((input) => {
+			// Expected ID: section-setting-input (e.g., "writer-ShowSidebar-input")
+			const parts = input.id.split('-');
+			if (parts.length !== 3 || parts[2] !== 'input') return;
+
+			const [sectionRaw, settingKey] = parts;
+			const value = input.checked;
+
+			if (sectionRaw === 'common') {
+				this.browserSettingOptions[settingKey] = value;
+			} else {
+				(this.browserSettingOptions[sectionRaw] as Record<string, boolean>)[
+					settingKey
+				] = value;
+			}
+		});
+	}
+
+	private customRenderers: Record<
+		string,
+		(key: string, value: any, uniqueId: string) => HTMLElement
+	> = {
+		compactToggle: this.renderCompactModeToggle.bind(this),
+	};
+
+	private renderCompactModeToggle(
+		key: string,
+		setting: any,
+		uniqueId: string,
+	): HTMLElement {
+		const container = document.createElement('div');
+		container.className = 'custom-compact-toggle';
+
+		const inputCheckbox = document.createElement('input');
+		inputCheckbox.type = 'checkbox';
+		inputCheckbox.className = 'checkbox-radio-switch-input';
+		inputCheckbox.id = uniqueId + '-input';
+		inputCheckbox.checked = setting.value;
+		inputCheckbox.style.display = 'none'; // hidden input for logic
+		container.appendChild(inputCheckbox);
+
+		const options = document.createElement('div');
+		options.className = 'toggle-options';
+
+		const select = (useCompact: boolean) => {
+			inputCheckbox.checked = useCompact;
+			setting.value = useCompact;
+			notebookImage.classList.toggle('selected', !useCompact);
+			compactImage.classList.toggle('selected', useCompact);
+		};
+
+		const notebookOption = this.createCompactToggleOption(
+			'Notebookbar.svg',
+			'Notebookbar',
+			_('Notebookbar view'),
+			!setting.value,
+			() => select(false),
+		);
+		const compactOption = this.createCompactToggleOption(
+			'Compact.svg',
+			'Compact',
+			_('Compact view'),
+			setting.value,
+			() => select(true),
+		);
+
+		const notebookImage = notebookOption.querySelector(
+			'.toggle-image',
+		) as HTMLImageElement;
+		const compactImage = compactOption.querySelector(
+			'.toggle-image',
+		) as HTMLImageElement;
+
+		options.appendChild(notebookOption);
```

Comment:
we can pass `options` to the `create` function and do it there for simplification